tools/userland-fetch

```diff
@@ -28,246 +28,371 @@
 #  directories, and verifies file hashes.
 #
 
+import errno
 import os
 import sys
 import shutil
+import subprocess
+import re
+import gzip
+import bz2
 from urllib import splittype
 from urllib2 import urlopen
 from urllib2 import Request
 import hashlib
 
+
```

Comment:
This is PEP8 specific, so I would rather leave it there. (PEP8 mandates to use two blank lines between imports and function/class definitions).